src/executor/thread_pool.cpp

```diff
@@ -10,86 +10,131 @@ namespace yaclib::executor {
 
 namespace {
 
-class ThreadPool final : public IThreadPool {
+thread_local IThreadPool* current_thread_pool;
```

Comment:
`tlCurrentThreadPool`/`tCurrentThreadPool`/`lCurrentThreadPool` thread local
`sCurrentThreadPool` static
`kCurrentThreadPool` constexpr/const static
`gCurrentThreadPool` global
What do you think?